### PR TITLE
Encapsulate http client with Axios

### PR DESCRIPTION
**Pull request by qingyanyang**
This PR sets up Axios client encapsulation and builds an example API encapsulation layer to provide a structured approach for managing API requests and services. It also establishes a service folder structure for better organization and maintainability.

Testing Method

- Simulated various error scenarios to validate interceptor behavior.
- Command 'npm run test'.

Related Ticket

[JIRA Ticket: IF-21](https://dext.atlassian.net/jira/software/c/projects/IF/boards/26?selectedIssue=IF-21)

### CODE REVIEW

File: src/services/httpClient.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  timeout: 30 * 1000,
+  responseType: "json",
+  headers: {
+    "Content-Type": "application/json"
+  }
+});
+
+// handle response
+appRequest.interceptors.response.use(
+  res => {
```
Comment (LH-Song) [inline src/services/httpClient.ts:16]: Did you guys have settings similar to .eslintrc.json? When using TypeScript, strict typing and data validation should be enabled. I noticed several places where strict request parameter types and return data types are needed.
```suggestion
  (res: AxiosResponse) => {
```

**Review by LH-Song — CHANGES_REQUESTED** [review]

File: src/services/httpClient.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  baseURL: API_BASE_URL,
+  timeout: 30 * 1000,
+  responseType: "json",
+  headers: {
+    "Content-Type": "application/json"
+  }
+});
+
+// handle response
+appRequest.interceptors.response.use(
```
[the remark this answers is not in the document]
Comment (qingyanyang) [reply src/services/httpClient.ts:15]: may refactor this in the future

File: src/services/httpClient.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import axios from "axios";
+import { API_BASE_URL } from "@/constants/apiConfig";
+import { ERROR_MESSAGES } from "@/constants/errorMessages";
+
+const appRequest = axios.create({
+  baseURL: API_BASE_URL,
+  timeout: 30 * 1000,
```
[the remark this answers is not in the document]
Comment (qingyanyang) [reply src/services/httpClient.ts]: I made all constants in one place

**Review by LH-Song — APPROVED** [review]
LGTM 🚀🚀

**Pull request merged**